components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/JITProvisioningPostAuthenticationHandler.java

```diff
@@ -289,28 +289,35 @@ private PostAuthnHandlerFlowStatus handleRequestFlow(HttpServletRequest request,
                     if (localClaimValues == null) {
                         localClaimValues = new HashMap<>();
                     }
+
+                    String associatedLocalUser =
+                            getLocalUserAssociatedForFederatedIdentifier(stepConfig.getAuthenticatedIdP(),
+                                    stepConfig.getAuthenticatedUser().getAuthenticatedSubjectIdentifier());
+
                     String username;
                     String userIdClaimUriInLocalDialect = getUserIdClaimUriInLocalDialect(externalIdPConfig);
                     if (isUserNameFoundFromUserIDClaimURI(localClaimValues, userIdClaimUriInLocalDialect)) {
                         username = localClaimValues.get(userIdClaimUriInLocalDialect);
                     } else {
-                        username = getLocalUserAssociatedForFederatedIdentifier(stepConfig.getAuthenticatedIdP(),
-                                stepConfig.getAuthenticatedUser().getAuthenticatedSubjectIdentifier());
+                        username = associatedLocalUser;
                     }
 
-                    // If username is null, that means relevant assication not exist already.
-                    if (StringUtils.isEmpty(username) && !isUserCreated) {
+                    // If associatedLocalUser is null, that means relevant association not exist already.
+                    if (StringUtils.isEmpty(associatedLocalUser) && !isUserCreated) {
```

Comment:
Noted below isUserCreated property is being set based on the username. Now as username also gets populated within this scope need to revisit and see if the condition validated there could still hold

Reply:
Yes, that can be a problem. Changed the logic to get the `getTenantDomainAppendedUserName(sequenceConfig.getAuthenticatedUser().getUserName(), context.getTenantDomain())` if `externalIdPConfig.isPromptConsentEnabled()` is enabled. In that case method should be returning and change to the username should not affect rest of the flow.